### PR TITLE
Fix undefined array indices while updating comments via API

### LINKED ISSUE #3687: Undefined indexes when moderating comments from WP.com notifications.
When I moderate a comment on a .org Jetpack site from within the notifications on a wp.com site, the error log for my .org site shows these warnings:

```
PHP Notice:  Undefined index: user_email in wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-comment-endpoint.php on line 84
PHP Notice:  Undefined index: status in wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-comment-endpoint.php on line 171
PHP Notice:  Undefined index: user_email in wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-comment-endpoint.php on line 84
```

The .org site is running the latest WP and Jetpack.

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes #3687

Fixes a bug checking for comment privileges, and ensures an array index exists when checking if an authenticated user can read a comment.

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-comment-endpoint.php [hunk 1]
```diff
@@ -81,6 +81,8 @@ function get_comment( $comment_id, $context ) {
 				&&
 					isset( $this->api->token_details['user'] )
 				&&
+					isset( $this->api->token_details['user']['user_email'] )
```
Comment (lezama) [inline json-endpoints/class.wpcom-json-api-comment-endpoint.php]: seeing this, shouldn't we check for `isset( $this->api->token_details['user']['display_name'] )` too?

File: json-endpoints/class.wpcom-json-api-update-comment-endpoint.php [hunk 2]
```diff
@@ -168,7 +168,7 @@ function update_comment( $path, $blog_id, $comment_id ) {
 		}
 
 		$comment_status = wp_get_comment_status( $comment->comment_ID );
-		if ( $comment_status !== $update['status'] && !current_user_can( 'moderate_comments' ) ) {
+		if ( $comment_status !== $update['comment_status'] && !current_user_can( 'moderate_comments' ) ) {
```
Comment (lezama) [inline json-endpoints/class.wpcom-json-api-update-comment-endpoint.php]: nice catch

**Comment by lezama** [comment]
Thanks! :shipit:  once tests pass.

File: json-endpoints/class.wpcom-json-api-comment-endpoint.php (hunk 1 above)
Comment (gravityrail) [inline json-endpoints/class.wpcom-json-api-comment-endpoint.php]: Probably, but that wasn't the error. Maybe that value is always set?

**Pull request merged**